submissions/AnnaStoyano/oop-exercise/index.js

```diff
@@ -0,0 +1,112 @@
+/* Refer to https://github.com/OleksiyRudenko/a-tiny-JS-world for the task details
+   Complete the below for code reviewers' convenience:
+
+   Code repository: _put repo URL here_
+   Web app: _put project's github pages URL here_
+   */
+
+// ======== OBJECTS DEFINITIONS ========
+// Define your objects here
+
+class Mammals{
+   constructor(name,gender,friends){
+      this.species = 'mammals',
+      this.name = name,
+      this.gender = gender,
+      this.legs = 4,
+      this.friends = friends
+   }
+}
+
+class Dog extends Mammals{
+   constructor(name,gender,friends,saying){
+      super(name,gender,friends);
+      this.species = 'dog',
+      this.saying = saying || 'guf'
```

Comment:
Every child class has property `saying`. This is a clear sign that it is a shared property and as such should be initialized in and handled by parent class.

Default value for saying can be defined at function definition. For exmaple `function abc(a, b, c = 5) { ... }`. Class method is basically a function. `constructor` is not an exclusion.